### PR TITLE
Cyb 478

### PR DESCRIPTION
**Pull request by justefg**

### CODE REVIEW

File: simulator/tests/randpa_finality_tests.cpp [hunk 1]
```diff
@@ -463,3 +463,57 @@ TEST(randpa_finality, no_threshold) {
...
+        }
+    }
+    
+    vector<pair<int, int> > v0{{1, 2}, {2, 10}};
+    graph_type g;
+    g.push_back(v0);
+    runner.load_graph(g);
+    
+    // dpos finality fails and randpa frozen
+    runner.add_stop_task(4000 * runner.get_slot_ms());
```
Comment (adrianopol) [inline simulator/tests/randpa_finality_tests.cpp]: Please, add a reference (comment) to the `_max_finality_lag_blocks` value here.

File: plugins/randpa_plugin/include/eosio/randpa_plugin/types.hpp [hunk 2]
```diff
@@ -25,7 +25,7 @@ using block_id_type = fc::sha256;
 using digest_type = fc::sha256;
 using signature_provider_type = std::function<signature_type(digest_type)>;
 
-uint32_t get_block_num(const block_id_type& id) {
+int32_t get_block_num(const block_id_type& id) {
```
[the remark this answers is not in the document]
Comment (justefg) [reply plugins/randpa_plugin/include/eosio/randpa_plugin/types.hpp]: a : uint32_t 
b : uint32_t 
if a < b then a - b overflows and that can cause some nasty bugs

[the remark this answers is not in the document]
Comment (adrianopol) [reply plugins/randpa_plugin/include/eosio/randpa_plugin/types.hpp]: Yes, but you can just make additional check in line 625.

[the remark this answers is not in the document]
Comment (justefg) [reply plugins/randpa_plugin/include/eosio/randpa_plugin/types.hpp]: well, what if I forget? actually, while developing this PR I forgot to add the check and spent 1-2 hours debugging the issue until it dawned on me this is an overflow

**Pull request merged**